YelpReview/resultViewController.swift

```diff
@@ -0,0 +1,235 @@
+//
+//  resultViewController.swift
+//  YelpReview
+//
+//  Created by Michael on 12/28/14.
+//  Copyright (c) 2014 Michael. All rights reserved.
+//
+
+import UIKit
+
+class resultViewController: UIViewController, FloatRatingViewDelegate {
+    
+    var stars: Int = 0;
+    var regenerations = 0;
+    var business: String = "";
+    var initialized = false;
+    var case1 = false
+    let loadingLabels = ["Currently :N: monkeys are working on writing your rewiew", "The server is connecting to the future through :W:", "Loading organic coffee to the server to generate a review", "Using organic bicycle energy to create this review", "Watching a Netflix series to get inspired for this one", "Feeding the server with kombucha", "Looking at :N: images of cute kittens for inspiration"]
+    let lw = ["matrix", "tv", "monkeys", "magic mongks", "kittens"]
+    
+    @IBOutlet weak var MonkeyFace: UIImageView!
+    
+    func getLoadingLabel() -> String {
+        var randomIndex = Int(arc4random_uniform(UInt32(self.loadingLabels.count)))
+        var randomLabel = self.loadingLabels[randomIndex]
+        randomIndex = Int(arc4random_uniform(UInt32(self.lw.count)))
+        var randomWord = self.lw[randomIndex]
+        let randomNumber = String(arc4random_uniform(UInt32(20)))
+        var result =
+        randomLabel.stringByReplacingOccurrencesOfString(":N:", withString: randomNumber, options: NSStringCompareOptions.LiteralSearch, range: nil).stringByReplacingOccurrencesOfString(":W:", withString: randomWord, options: NSStringCompareOptions.LiteralSearch, range: nil)
+        return result
+    }
+    
+    @IBOutlet var floatRatingView: FloatRatingView!
+    @IBOutlet weak var businessType: UILabel!
+    //@IBOutlet weak var stars: UILabel!
+    @IBOutlet weak var resultText: UITextView!
+    @IBOutlet weak var generateButton: UIButton!
+    @IBOutlet weak var resultContainer: UIView!
+    var approaches = 3
+    var back: UIImageView!
+    var front: UIImageView!
+    var result: UITextView!
+    var loading = false
+    var shareText = ""
+    var shareUrl = ""
+    @IBOutlet weak var loadingView: UIView!
+    
+    @IBOutlet weak var loadingTextView: UITextView!
+    
+    func startLoading() {
+        self.loading = true
+        self.generateButton.hidden = true
+        self.shareView.hidden = true
+        self.loadingView.hidden = false
+        self.loadingTextView.text = getLoadingLabel()
+        self.loadingTextView.font = UIFont(name: "Helvetica", size: 16)
+        self.MonkeyFace.alpha = 1
+
+
+    }
+    
+    func stopLoading() {
+        if self.regenerations == self.approaches {
+            self.generateButton.setTitle("New review", forState: UIControlState.Normal)
```

Comment:
If you have time it would be better to subclass this generate button and change its appearance based on a enum property such as "NewReviewState"